File: csharp/Gherkin/Ast/DataTable.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Gherkin.Ast
{
    public class DataTable : StepArgument, IHasRows
    {
        public IEnumerable<TableRow> Rows { get; private set; }

        public DataTable(TableRow[] rows)
        {
            Rows = rows;
        }
    }
}

File: java/src/main/java/gherkin/ast/DataTable.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class DataTable extends Node {
    private final List<TableRow> rows;

    public DataTable(List<TableRow> rows) {
        super(rows.get(0).getLocation());
        this.rows = Collections.unmodifiableList(rows);
    }

    @Override
    public void accept(Visitor visitor) {
        for (TableRow row : rows) {
            row.accept(visitor);
        }
        visitor.visitDataTable(this);
    }
}
